ReviewBot.py

```diff
@@ -375,6 +379,63 @@ def set_request_ids_project(self, project, typename):
             req.read(request)
             self.requests.append(req)
 
+    def comment_handler_add(self, level=logging.INFO):
+        """Add handler to start recording log messages for comment."""
+        self.comment_handler = CommentFromLogHandler(level)
+        self.logger.addHandler(self.comment_handler)
+
+    def comment_handler_remove(self):
+        self.logger.removeHandler(self.comment_handler)
+
+    def comment_find(self, request=None, state=None, result=None):
+        """Return previous comments by current bot and matching criteria."""
+        # Case-insensitive for backwards compatibility.
+        bot = self.__class__.__name__.lower()
```

Comment:
That was intentional since the camelCasing used in class names makes them easier to read if the comment is ever looked at. Extremely minor, but the case here is `lower()` just like the match group to ensure backwards compatibility with old-style `leaper` as noted in original description. If desired I can make the new code print lower case as well.